apis/v1beta1/types.go

```diff
@@ -17,6 +17,10 @@ type ProviderConfigSpec struct {
 	// AssumeRoleChain defines the options for assuming an IAM role
 	AssumeRoleChain []AssumeRoleOptions `json:"assumeRoleChain,omitempty"`
 
+	// AssumeRoleSTSRegion defines the region used for STS requests. If not defined,
+	// the managed resource region will be used if available
+	AssumeRoleSTSRegion *string `json:"assumeRoleSTSRegion,omitempty"`
```

Comment:
An alternative to consider would be to add the the STS region option to the `v1beta1.AssumeRoleOptions` type. Then, we the `ProviderConfig` would look like:
```yaml
...
spec:
  assumeRoleChain:
  - stsRegion: us-west-1
    roleARN: <role ARN 1>
  - stsRegion: us-west-1
    roleARN: <role ARN 2>
  credentials:
    secretRef:
      key: creds
      name: aws-creds
      namespace: upbound-system
    source: Secret
```

However, you'd need to configure an STS region for each role in the chain if you are provisioning a resource without a region (e.g., IAM role) and not sure if there's utility in configuring different STS regions for different roles in the chain. Of course, if needed, we could have both levels of API exposed, e.g., a `spec.assumeRoleSTSRegion` and `spec.assumeRoleChain[].stsRegion` (which we could add later if needed). So, I think the proposed API is fine. What do you think?

Reply:
my first idea was also to put it in `v1beta1.AssumeRoleOptions`, but then I moved it because I didn't think there would be a need to chain assume role requests across different regions. I definitely would have rather put it inside an assumeRole config, but the current structure doesn't really support that. so I think this is probably the best option

Reply:
As a side note for documentation purposes, we had also previously discussed defaulting to a region specified in the `ProviderConfig` if a managed resource does not specify an optional `spec.forProvider.region`. The discussion's context was not this PR's target issue (but rather the UX around provider-aws MR APIs). But looks like it's also related to the current discussion.